umm-spec-lib/src/cmr/umm_spec/xml_to_umm_mappings/dif9/spatial_extent.clj

```diff
@@ -0,0 +1,127 @@
+  [doc bounding-box-map]
+  (remove nil?
+    (for [el (select doc "/DIF/Spatial_Coverage")]
+      (build-bounding-rectangle el bounding-box-map nil))))
+
+(defn- parse-horizontal-domain
+  "Create the horizontal spatial domain part of the spatial extent."
+  [doc]
+  (when-let [brs (seq (create-bounding-rectangles doc bounding-box-map))]
+    {:HorizontalSpatialDomain
+     {:Geometry {:CoordinateSystem "CARTESIAN" ;; DIF9 doesn't have CoordinateSystem, default to CARTESIAN
+                 :BoundingRectangles brs}}}))
+
+;; Vertical Domain parsing
+;; This is the list to go through to get all of the DIF 9 elevation elements.
+(def elevation-list ["Minimum_Altitude" "Maximum_Altitude" "Minimum_Depth" "Maximum_Depth"])
+
+(defn- create-vertical-domain-obj
+  "This method returns a vertical domain map that consists of {:Type type :Value value} if a value exists
+  otherwise it returns nil."
+  [spatial-coverage elevation-type]
+  (if-let [value (value-of spatial-coverage elevation-type)]
+    (into {}
+       {:Type (string/replace elevation-type "_" " ")
+        :Value value})))
+
+(defn- build-vertical-domains
+  "This function uses recursion to peice together all of the vertical domain maps into
+   a vector without nil values. The return value is a vector of either maps or its empty.
+   [{:Type \"Minimum Altitude\", :Value \"0\"}
+    {:Type \"Maximum Altitude\", :Value \"100\"}]
+   Nil values will not exist in the vector."
+  [spatial-coverage elevation-list result]
+  (if-let [elevation-type (first elevation-list)]
+    (if (= elevation-type "Maximum_Depth")
+      (if-let [obj (create-vertical-domain-obj spatial-coverage elevation-type)]
+        (conj result obj)
+        result)
+      (if-let [obj (create-vertical-domain-obj spatial-coverage elevation-type)]
+        (let [new-result (conj result obj)]
+          (build-vertical-domains spatial-coverage (drop 1 elevation-list) new-result))
+        (build-vertical-domains spatial-coverage (drop 1 elevation-list) result)))
+    result))
+
+(defn- create-vertical-domains
+  "Returns a seq of a vector that contains vertical spatial domains by the given DIF XML.
+   The DIF XML is passed in."
+ [doc]
+ (for [spatial_coverage (select doc "/DIF/Spatial_Coverage")]
+   (build-vertical-domains spatial_coverage elevation-list [])))
+
+(defn- parse-vertical-domains
+  "Create the vertical spatial domains part of the spatial extent."
+  [doc]
+  (when-let [vertical-domains (vec (flatten (create-vertical-domains doc)))]
+    (if (not-empty vertical-domains)
+      {:VerticalSpatialDomains vertical-domains})))
+
+(defn- parse-granule-representation
+  "Parse the granule spatial representation"
+  [doc sanitize?]
+  {:GranuleSpatialRepresentation (or (value-of doc "/DIF/Extended_Metadata/Metadata[Name='GranuleSpatialRepresentation']/Value")
+                                     (when sanitize?
+                                       "NO_SPATIAL"))})
```

Comment:
The umm-spec utils file already had a "NO_SPATIAL" default value.  I used it in place of "NO_SPATIAL".